File: pinyougou-interface/src/main/java/cn/itcast/core/service/OrderService.java
package cn.itcast.core.service;

import cn.itcast.core.entity.PageResult;
import cn.itcast.core.pojo.order.Order;

import java.util.List;

public interface OrderService {
    void add(Order order, String name);

    List<Order> findAll(String name);

    Order queryBrandByOrderId(Long orderId);

    PageResult search(int page, int rows, Order order, String name);

    void updateStatus(Long[] ids, String status);

}

File: pinyougou-web-shop/src/main/webapp/js/service/orderService.js
//服务层
app.service('orderService',function($http){

    //读取列表数据绑定到表单中
    this.findAll=function () {
        return $http.get('../order/findAll.do');
    }

    // 根据Id查询
    //根据Id查询
    this.queryBrandByOrderId=function (orderId) {
        return $http.get("../order/queryBrandByOrderId.do?orderId=" + orderId);
    }

    //搜索
    this.search=function(page,rows,searchEntity){
        return $http.post('../order/search.do?page='+page+"&rows="+rows, searchEntity);
    }

    // 发货
    this.updateStatus = function(ids,status){
        return $http.get('../order/updateStatus.do?ids='+ids+"&status="+status);
    }

});
